src/yaml/aslYamlLanguageService.ts

```diff
@@ -30,8 +31,40 @@ import {
 import { YAMLCompletion } from 'yaml-language-server/out/server/src/languageservice/services/yamlCompletion'
 import { YAMLSchemaService } from 'yaml-language-server/out/server/src/languageservice/services/yamlSchemaService'
 import { matchOffsetToDocument } from 'yaml-language-server/out/server/src/languageservice/utils/arrUtils'
+import { YAMLDocDiagnostic } from 'yaml-language-server/out/server/src/languageservice/utils/parseUtils'
 import doCompleteAsl from '../completion/completeAsl'
 
+function convertYAMLDiagnostic(yamlDiagnostic: YAMLDocDiagnostic, textDocument: TextDocument): Diagnostic {
+    const startLoc = yamlDiagnostic.location.start
+    let endLoc = yamlDiagnostic.location.end
+    let severity = yamlDiagnostic.severity
+
+    // Duplicate positining returns incorect end position and needs to be ovewritten
+    if (yamlDiagnostic.message === 'duplicate key') {
+        const text = textDocument.getText()
+        // Update severity to error
+        severity = 1
+
+        for (let loc = yamlDiagnostic.location.start; loc < text.length; loc++) {
+            // Colon and whitespace character signal end of the key.
+            if (text.slice(loc, loc + 2).match(/:\s/)) {
```

Comment:
Doesn't this fail index out of bounds if we have loc = text.length - 1?

Reply:
No, slice will slice until the last character of the string (including it) if the upper bound is too high.

Reply:
Didn't know that, ignore the comment then :)